src/chunk.c

```diff
@@ -4267,3 +4118,66 @@ ts_chunk_status(PG_FUNCTION_ARGS)
 	Chunk *chunk = ts_chunk_get_by_relid(chunk_relid, /* fail_if_not_found */ true);
 	PG_RETURN_INT32(chunk->fd.status);
 }
+
+/*
+ * Lock the chunk if the lockmode demands it.
+ *
+ * Also check that the chunk relation actually exists after the lock is
+ * acquired. Return true if no locking is necessary or the chunk relation
+ * exists and the lock was successfully acquired. Otherwise return false.
+ */
+bool
+ts_chunk_lock_if_exists(Oid chunk_oid, LOCKMODE chunk_lockmode)
+{
+	/* No lock is requests, so assume relation exists */
+	if (chunk_lockmode == NoLock)
+		return true;
```

Comment:
We never call this function with NoLock in our code, but the line should probably be here for completeness. (This is also based on corresponding locking done in PG's `find_inheritance_children()`

Reply:
I'd use `Assert(false)` for the stuff that should never happen, but maybe it's fine this way too.

Reply:
I inverted the check instead, which makes the logic equivalent but codecov should stop complaining.

We only use AccessShareLock currently, so other lockmodes not tested either.